shared/src/utils/parsePseudoCode.ts

```diff
@@ -0,0 +1,555 @@
+  const breakLatex = "break"
+  return { breakNormal, breakColor, breakLatex }
+}
+
+function pseudoCodeCallToString(call: PseudoCodeCall): {
+  callNormal: string
+  callColor: string
+  callLatex: string
+} {
+  let callNormal = `\\textit{${call.functionName} }(`
+  let callColor = `{\\color{${functionColor}} \\textit{${call.functionName} }}(`
+  let callLatex = `\\Call{${call.functionName}}{`
+
+  for (let i = 0; i < call.args.length; i++) {
+    const { psString, psStringColor, psStringLatex } = pseudoCodeStringToString(call.args[i])
+    callNormal += psString
+    callColor += psStringColor
+    callLatex += psStringLatex
+    if (i < call.args.length - 1) {
+      callNormal += ", "
+      callColor += ", "
+      callLatex += ", "
+    }
+  }
+
+  callNormal += ")"
+  callColor += ")"
+  callLatex += "}"
+
+  return { callNormal, callColor, callLatex }
+}
+
+function pseudoCodeStringToString(pseudoString: PseudoCodeString): {
+  psString: string
+  psStringColor: string
+  psStringLatex: string
+} {
+  let psString = ""
+  let psStringColor = ""
+  let psStringLatex = ""
+  for (let i = 0; i < pseudoString.length; i++) {
+    if (typeof pseudoString[i] === "string") {
+      psString += pseudoString[i] as string
+      psStringColor += pseudoString[i] as string
+      psStringLatex += pseudoString[i] as string
+    } else if (isPseudoCodeVariable(pseudoString[i])) {
+      const { variable } = pseudoString[i] as PseudoCodeVariable
+      psString += variable
+      psStringColor += `{\\color{${variableColor}}${variable}}`
+      psStringLatex += variable
+    } else if (isPseudoCodeFunctionName(pseudoString[i])) {
+      const { functionName } = pseudoString[i] as PseudoCodeFunctionName
+      psString += functionName
+      psStringColor += `{\\color{${functionColor}}${functionName}}`
+      psStringLatex += functionName
+    } else if (isPseudoCodePrintString(pseudoString[i])) {
+      const { printString } = pseudoString[i] as PseudoCodePrintString
+      psString += printString
+      psStringColor += `{\\color{${printStatementColor}}${printString}}`
+      psStringLatex += printString
+    }
+  }
+
+  return { psString, psStringColor, psStringLatex }
+}
+
+function isPseudoCodeVariable(
+  pseudoString: string | PseudoCodeVariable | PseudoCodeFunctionName | PseudoCodePrintString,
+): boolean {
+  return typeof pseudoString === "object" && "variable" in pseudoString
+}
+
+function isPseudoCodeFunctionName(
+  pseudoString: string | PseudoCodeVariable | PseudoCodeFunctionName | PseudoCodePrintString,
+): boolean {
+  return typeof pseudoString === "object" && "functionName" in pseudoString
+}
+
+function isPseudoCodePrintString(
+  pseudoString: string | PseudoCodeVariable | PseudoCodeFunctionName | PseudoCodePrintString,
+): boolean {
+  return typeof pseudoString === "object" && "printString" in pseudoString
+}
+
+function leadingWhitespace(indent: number): string {
```

Comment:
for the latex part it would probably be better to use something like `\hspace{}` instead of whitespaces. Latex has the option to shrink the size of whitespace if the line is too full.

Reply:
I would suggested I take this out from the Latex part, because the package `Algpseudocodex` does indentation  automatically.